File: susupay-backend/app/schemas/report.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class ClientSummaryItem(BaseModel):
    client_name: str
    total_deposits: Decimal
    deposit_count: int
    total_payouts: Decimal
    payout_count: int
    net_balance: Decimal


class MonthlySummary(BaseModel):
    year: int
    month: int
    total_deposits: Decimal
    total_payouts: Decimal
    net_balance: Decimal
    client_count: int
    clients: list[ClientSummaryItem]


class ClientStatementItem(BaseModel):
    date: datetime
    type: str  # DEPOSIT | PAYOUT
    description: str
    amount: Decimal
    running_balance: Decimal


class ClientStatement(BaseModel):
    client_name: str
    year: int
    month: int
    opening_balance: Decimal
    closing_balance: Decimal
    items: list[ClientStatementItem]

File: susupay-frontend/src/types/report.ts
export interface ClientSummaryItem {
  client_id: string;
  client_name: string;
  total_deposits: string;
  deposit_count: number;
  total_payouts: string;
  payout_count: number;
  net_balance: string;
}

export interface MonthlySummary {
  year: number;
  month: number;
  total_deposits: string;
  total_payouts: string;
  net_balance: string;
  client_count: number;
  clients: ClientSummaryItem[];
}

export type StatementEntryType = 'DEPOSIT' | 'PAYOUT';

export interface ClientStatementItem {
  date: string;
  type: StatementEntryType;
  description: string;
  amount: string;
  running_balance: string;
}

export interface ClientStatement {
  client_name: string;
  year: number;
  month: number;
  opening_balance: string;
  closing_balance: string;
  items: ClientStatementItem[];
}
